src/main/java/com/microsoft/sqlserver/jdbc/SQLServerCallableStatement.java

```diff
@@ -150,6 +151,21 @@ public void registerOutParameter(int index, int sqlType) throws SQLServerExcepti
             case microsoft.sql.Types.DATETIMEOFFSET:
                 param.setOutScale(7);
                 break;
+            case java.sql.Types.DECIMAL:
+                // Dynamically handle the scale for DECIMAL output parameters.
+                // The scale for the DECIMAL type is fetched from the ParameterMetaData.
+                // This provides flexibility to automatically apply the correct scale as per the database metadata.
+                ParameterMetaData parameterMetaData = this.getParameterMetaData();
+                if (parameterMetaData != null) {
+                    try {
+                        // Fetch scale from metadata for DECIMAL type
+                        int scale = parameterMetaData.getScale(index);
+                        param.setOutScale(scale);
+                    } catch (SQLException e) {
+                        loggerExternal.warning("Failed to fetch scale for DECIMAL type parameter at index " + index + ": " + e.getMessage());
```

Comment:
should this fail? according to the spec this would only throw an error if a database access error occurs, seem quite a serious error